src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/RecommenderPackageFeed.cs

```diff
@@ -0,0 +1,192 @@
+
+            if (dependentPackages == null)
+            {
+                throw new ArgumentNullException(nameof(dependentPackages));
+            }
+            _dependentPackages = dependentPackages;
+
+            if (targetFrameworks == null)
+            {
+                throw new ArgumentNullException(nameof(targetFrameworks));
+            }
+            _targetFrameworks = targetFrameworks;
+
+            if (metadataProvider == null)
+            {
+                throw new ArgumentNullException(nameof(metadataProvider));
+            }
+            _metadataProvider = metadataProvider;
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+            _logger = logger;
+
+            // Get NuGet package recommender service, or null if it is not available
+            NuGetRecommender = Package.GetGlobalService(typeof(SVsNuGetRecommenderService)) as IVsNuGetPackageRecommender;
+            if(NuGetRecommender != null)
+            {
+                var VersionDict = NuGetRecommender.GetVersionInfo();
+                VersionInfo.modelVersion = VersionDict.ContainsKey("Model") ? VersionDict["Model"] : null;
+                VersionInfo.vsixVersion = VersionDict.ContainsKey("Vsix") ? VersionDict["Vsix"] : null;
+            }
+        }
+
+        private class RecommendSearchToken : ContinuationToken
+        {
+            public string SearchString { get; set; }
+            public SearchFilter SearchFilter { get; set; }
+        }
+
+        public Task<SearchResult<IPackageSearchMetadata>> SearchAsync(string searchText, SearchFilter searchFilter, CancellationToken cancellationToken)
+        {
+            var searchToken = new RecommendSearchToken
+            {
+                SearchString = searchText,
+                SearchFilter = searchFilter,
+            };
+
+            return RecommendPackagesAsync(searchToken, cancellationToken);
+        }
+
+        public async Task<SearchResult<IPackageSearchMetadata>> RecommendPackagesAsync(ContinuationToken continuationToken, CancellationToken cancellationToken)
+        {
+            var searchToken = continuationToken as RecommendSearchToken;
+            if (searchToken == null)
+            {
+                throw new InvalidOperationException("Invalid token");
+            }
+            // don't make recommendations if the user entered a search string
+            if (searchToken.SearchString != "")
+            {
+                return SearchResult.Empty<IPackageSearchMetadata>();
+            }
+
+            List<string> recommendIds = new List<string>();
+            if (NuGetRecommender != null)
+            {
+                // get lists of only the package ids to send to the recommender
+                List<string> topPackages = _installedPackages.Select(item => item.Id.ToLowerInvariant()).ToList();
+                List<string> depPackages = _dependentPackages.Keys.ToList();
```

Comment:
Wouldn't we want to eliminate duplicates, here? Anyway, this is all dead code since we're not implementing PC... not sure if it should all be commented out or removed or left in place.

Reply:
I removed the rest of the dependent packages code. All I left is creating an empty list of package ids to send to the recommender, since the recommender still expects this parameter, and is able to use dependent packages in recommendations when available.